model/file.go

```diff
@@ -0,0 +1,158 @@
+package model
+
+import (
+	"errors"
+	"github.com/unidoc/unipdf/v3/common"
+	"github.com/unidoc/unipdf/v3/core"
+)
+
+// (Section 7.11.3 p. 102).
+// See Table 44 - Entries in a file specification dictionary
+
+// PdfFilespec represents a file specification
+type PdfFilespec struct {
+	Type core.PdfObject
+	FS   core.PdfObject
+	F    core.PdfObject // A file specification string
+	UF   core.PdfObject //A Unicode text string that provides file specification
+	DOS  core.PdfObject // A file specification string representing a DOS file name. OBSOLETE
+	Mac  core.PdfObject // A file specification string representing a Mac OS file name. OBSOLETE
+	Unix core.PdfObject // A file specification string representing a UNIX file name. OBSOLETE
+	ID   core.PdfObject // An array of two byte strings constituting a file identifier
+	V    core.PdfObject // A flag indicating whether the file referenced by the file specification is volatile (changes frequently with time).
+	EF   core.PdfObject // A dictionary containing a subset of the keys F, UF, DOS, Mac, and Unix, corresponding to the entries by those names in the file specification dictionary
+	RF   core.PdfObject
+	Desc core.PdfObject // Descriptive text associated with the file specification
+	CI   core.PdfObject // A collection item dictionary, which shall be used to create the user interface for portable collections
+
+	container core.PdfObject
+}
+
+// GetContainingPdfObject implements interface PdfModel.
+func (f *PdfFilespec) GetContainingPdfObject() core.PdfObject {
+	return f.container
+}
+
+func (f *PdfFilespec) getDict() *core.PdfObjectDictionary {
+	if indObj, is := f.container.(*core.PdfIndirectObject); is {
+		dict, ok := indObj.PdfObject.(*core.PdfObjectDictionary)
+		if !ok {
+			return nil
+		}
+		return dict
+	} else if dictObj, isDict := f.container.(*core.PdfObjectDictionary); isDict {
+		return dictObj
+	} else {
+		common.Log.Debug("Trying to access Filespec dictionary of invalid object type (%T)", f.container)
+		return nil
+	}
+}
+
+// ToPdfObject implements interface PdfModel.
+func (f *PdfFilespec) ToPdfObject() core.PdfObject {
+	d := f.getDict()
+
+	d.Clear()
+
+	d.Set("Type", core.MakeName("Filespec"))
+	d.SetIfNotNil("FS", f.FS)
+	d.SetIfNotNil("F", f.F)
+	d.SetIfNotNil("UF", f.UF)
+	d.SetIfNotNil("DOS", f.DOS)
+	d.SetIfNotNil("Mac", f.Mac)
+	d.SetIfNotNil("Unix", f.Unix)
+	d.SetIfNotNil("ID", f.ID)
+	d.SetIfNotNil("V", f.V)
+	d.SetIfNotNil("EF", f.EF)
+	d.SetIfNotNil("RF", f.RF)
+	d.SetIfNotNil("Desc", f.Desc)
+	d.SetIfNotNil("CI", f.CI)
+
+	return f.container
+}
+
+// NewPdfFilespecFromObj creates and returns a new PdfFilespec object.
+func NewPdfFilespecFromObj(obj core.PdfObject) (*PdfFilespec, error) {
+	fs := &PdfFilespec{}
+
+	var dict *core.PdfObjectDictionary
+	if indObj, isInd := obj.(*core.PdfIndirectObject); isInd {
```

Comment:
Should use `core.GetIndirect` and `core.GetDict` here in order to resolve potential references:

```go
var dict *core.PdfObjectDictionary
if indObj, isInd := core.GetIndirect(obj); isInd {
	fs.container = indObj

	d, ok := core.GetDict(indObj.PdfObject)
	if !ok {
		common.Log.Debug("Object not a dictionary type")
		return nil, core.ErrTypeError
	}
	dict = d
} else if d, isDict := core.GetDict(obj); isDict {
	fs.container = d
	dict = d
} else {
	common.Log.Debug("Object type unexpected (%T)", obj)
	return nil, core.ErrTypeError
}
```